--- RelationalCool/cmt/requirements.orig	2015-02-14 18:50:39.211804857 +0100
+++ RelationalCool/cmt/requirements	2015-02-14 18:51:21.079859733 +0100
@@ -37,6 +37,9 @@
 
 use CppUnit v* LCG_Interfaces -no_auto_imports
 
+macro_prepend CppUnit_linkopts " -Wl,--no-as-needed "
+macro_append CppUnit_linkopts " -Wl,--as-needed "
+
 # Unit tests with no database connection
 apply_pattern cool_unit_test tname=ChannelSelection
 apply_pattern cool_unit_test tname=PayloadSpecification
