--- CoralCommon/cmt/requirements	(revisione 19531)
+++ CoralCommon/cmt/requirements	(copia locale)
@@ -20,6 +20,9 @@
 
 use CppUnit v* LCG_Interfaces -no_auto_imports
 
+macro_prepend CppUnit_linkopts " -Wl,--no-as-needed "
+macro_append CppUnit_linkopts " -Wl,--as-needed "
+
 # Do not install the header files (task #15556)
 macro_remove constituents 'install_includes'
 
Index: CoralKernel/cmt/requirements
===================================================================
--- CoralKernel/cmt/requirements	(revisione 19531)
+++ CoralKernel/cmt/requirements	(copia locale)
@@ -17,8 +17,13 @@
 
 private
 
+macro_append lcg_CoralKernel_use_linkopts " -ldl "
+
 use CppUnit v* LCG_Interfaces -no_auto_imports
 
+macro_prepend CppUnit_linkopts " -Wl,--no-as-needed "
+macro_append CppUnit_linkopts " -Wl,--as-needed "
+
 # The unit tests
 apply_pattern coral_unit_test tname=CoralKernelTest
 apply_pattern coral_unit_test tname=ExternalPluginManager
