--- cmake/FindROOT.cmake.orig	2014-04-02 15:03:44.000000000 +0200
+++ cmake/FindROOT.cmake	2014-04-02 15:04:14.000000000 +0200
@@ -9,7 +9,7 @@
 
 
 find_program(ROOT_CONFIG_EXECUTABLE root-config
-  PATHS $ENV{ROOTSYS}/bin)
+  PATHS ${ROOTSYS}/bin $ENV{ROOTSYS}/bin)
 
 if(ROOT_CONFIG_EXECUTABLE)
   execute_process(
@@ -56,7 +56,7 @@
 mark_as_advanced(ROOT_FOUND ROOT_INCLUDE_DIR)
 
 include(CMakeParseArguments)
-find_program(ROOTCINT_EXECUTABLE rootcint PATHS $ENV{ROOTSYS}/bin)
+find_program(ROOTCINT_EXECUTABLE rootcint PATHS ${ROOTSYS}/bin $ENV{ROOTSYS}/bin)
 
 #----------------------------------------------------------------------------
 # function root_generate_dictionary( dictionary   
