--- requirements	2013-11-26 10:24:16.000000000 +0100
+++ requirements.new	2013-12-01 21:55:18.000000000 +0100
@@ -7,9 +7,9 @@
       ATLAS&NIGHTLIES "/afs/cern.ch/sw/lcg/app/nightlies/$(LCG_NGT_SLT_NAME)/$(LCG_NGT_DAY_NAME)" \
       ATLAS           "${SITEROOT}/sw/lcg/app/releases"
 
-macro LCG_external       "$(LCG_Settings_root)/../../../../../external" \
+macro LCG_external       "$(LCG_Settings_root)/../../.." \
       ATLAS&experimental "${SITEROOT}/sw/lcg/experimental" \
-      experimental       "$(LCG_Settings_root)/../../../../../experimental" \
+      experimental       "$(LCG_Settings_root)/../../.." \
       ATLAS              "${SITEROOT}/sw/lcg/external"
 
 # ==== pattern to express optional dependencies =====
