index eca73e9..40e3ae8 100644
--- cint/reflex/CMakeLists.txt
+++ cint/reflex/CMakeLists.txt
@@ -15,7 +15,7 @@ if(CMAKE_PROJECT_NAME STREQUAL ROOT)
                            Reflex/SharedLibrary.h Reflex/DictionaryGenerator.h)
   #---Rootcint dictionary ---------------------------------------------------
   ROOT_GENERATE_DICTIONARY(G__Reflex ${headers} LINKDEF LinkDef.h OPTIONS -p)
-  ROOT_GENERATE_ROOTMAP(Reflex LINKDEF LinkDef.h DEPENDENCIES )
+  ROOT_GENERATE_ROOTMAP(ReflexDict LINKDEF LinkDef.h DEPENDENCIES Reflex )
   ROOT_MODULE_LIBRARY(ReflexDict G__Reflex.cxx LIBRARIES Reflex Cint Core)
   #---Reflex library---------------------------------------------------------
   ROOT_LINKER_LIBRARY(Reflex *.cxx LIBRARIES ${CMAKE_DL_LIBS})  
