diff --git a/net/netxng/src/TNetXNGFile.cxx b/net/netxng/src/TNetXNGFile.cxx
index e99767d..613b6fa 100644
--- net/netxng/src/TNetXNGFile.cxx
+++ net/netxng/src/TNetXNGFile.cxx
@@ -403,7 +403,7 @@ Bool_t TNetXNGFile::ReadBuffers(char *buffer, Long64_t *position, Int_t *length,
    std::vector<XRootDStatus*> *statuses;
    TSemaphore                 *semaphore;
    Int_t                       totalBytes = 0;
-   Int_t                       offset     = 0;
+   Long64_t                    offset     = 0;
 
    Double_t start = 0;
    if (gPerfStats) start = TTimeStamp();
@@ -481,13 +481,13 @@ Bool_t TNetXNGFile::ReadBuffers(char *buffer, Long64_t *position, Int_t *length,
 
       if (!st->IsOK()) {
          Error("ReadBuffers", "%s", st->ToStr().c_str());
-         delete statuses;
-         delete semaphore;
          for( ; it != chunkLists.end(); ++it )
          {
             st = statuses->at( it - chunkLists.begin() );
             delete st;
          }
+         delete statuses;
+         delete semaphore;
 
          return kTRUE;
       }
diff --git a/net/netxng/src/TNetXNGFile.cxx b/net/netxng/src/TNetXNGFile.cxx
index 9c1f4da..c28b18c 100644
--- net/netxng/src/TNetXNGFile.cxx
+++ net/netxng/src/TNetXNGFile.cxx
@@ -404,6 +404,7 @@ Bool_t TNetXNGFile::ReadBuffers(char *buffer, Long64_t *position, Int_t *length,
    TSemaphore                 *semaphore;
    Int_t                       totalBytes = 0;
    Long64_t                    offset     = 0;
+   char                       *cursor     = buffer;
 
    Double_t start = 0;
    if (gPerfStats) start = TTimeStamp();
@@ -425,16 +426,17 @@ Bool_t TNetXNGFile::ReadBuffers(char *buffer, Long64_t *position, Int_t *length,
          // Add as many max-size chunks as are divisible
          for (j = 0; j < nsplit; ++j) {
             offset = position[i] + (j * fReadvIorMax);
-            chunks.push_back(ChunkInfo(offset, fReadvIorMax, buffer));
+            chunks.push_back(ChunkInfo(offset, fReadvIorMax, cursor));
+            cursor += fReadvIorMax;
          }
 
          // Add the remainder
          offset = position[i] + (j * fReadvIorMax);
-         chunks.push_back(ChunkInfo(offset, rem, buffer));
-
+         chunks.push_back(ChunkInfo(offset, rem, cursor));
+         cursor += rem;
       } else {
-         offset = position[i];
-         chunks.push_back(ChunkInfo(offset, length[i], buffer));
+         chunks.push_back(ChunkInfo(position[i], length[i], cursor));
+         cursor += length[i];
       }
 
       // If there are more than or equal to max chunks, make another chunk list
@@ -449,7 +451,8 @@ Bool_t TNetXNGFile::ReadBuffers(char *buffer, Long64_t *position, Int_t *length,
    }
 
    // Push back the last chunk list
-   chunkLists.push_back(chunks);
+   if( !chunks.empty() )
+      chunkLists.push_back(chunks);
 
    TAsyncReadvHandler *handler;
    XRootDStatus        status;
@@ -462,7 +465,7 @@ Bool_t TNetXNGFile::ReadBuffers(char *buffer, Long64_t *position, Int_t *length,
    {
       handler = new TAsyncReadvHandler(statuses, it - chunkLists.begin(),
                                        semaphore);
-      status  = fFile->VectorRead(*it, buffer, handler);
+      status = fFile->VectorRead(*it, 0, handler);
 
       if (!status.IsOK()) {
          Error("ReadBuffers", "%s", status.ToStr().c_str());
diff --git a/net/netxng/src/TNetXNGFile.cxx b/net/netxng/src/TNetXNGFile.cxx
index c28b18c..65824b7 100644
--- net/netxng/src/TNetXNGFile.cxx
+++ net/netxng/src/TNetXNGFile.cxx
@@ -576,7 +576,7 @@ XrdCl::OpenFlags::Flags TNetXNGFile::ParseOpenMode(Option_t *modestr)
    // returns:      correctly parsed option mode
 
    using namespace XrdCl;
-   OpenFlags::Flags mode = OpenFlags::None;
+   OpenFlags::Flags mode = OpenFlags::Read;
    TString mod = ToUpper(TString(modestr));
 
    if (mod == "NEW" || mod == "CREATE")  mode = OpenFlags::New;
