--- setup.py.org	2008-09-01 12:42:14.000000000 +0200
+++ setup.py	2010-05-06 11:30:01.000000000 +0200
@@ -4,13 +4,8 @@
 import os, subprocess
 
 ##################################################################################################
-try :
-    incdir=os.popen('root-config --incdir').read()
-    incdir=incdir.strip('\n')
-    libdir=os.popen('root-config --libdir').read()
-    libdir=libdir.strip('\n')
-except RuntimeError:
-    print "root-config not found in PATH"
+incdir="/afs/.cern.ch/sw/lcg/external/minuit/"+os.environ["VERSION"]+"/"+os.environ["CMTCONFIG"]+"/include"
+libdir="/afs/.cern.ch/sw/lcg/external/minuit/"+os.environ["VERSION"]+"/"+os.environ["CMTCONFIG"]+"/lib"
     
 setup(name="pyMinuit2",
       version="0.0.1",
@@ -23,6 +18,6 @@
                              [os.path.join("minuit2.cpp")],
                              library_dirs=[libdir],
                              
-libraries=["Core","Cint","RIO","Net","Hist","Graf","Rint","Matrix","MathCore","Minuit2"],
+libraries=["Minuit2"],
                              include_dirs=[incdir]
                              )])