--- src/crmchep.h.orig	2013-04-26 20:04:19.000000000 +0200
+++ src/crmchep.h	2013-04-26 20:05:07.000000000 +0200
@@ -16,6 +16,7 @@
 #include <HepMC/Units.h>
 #endif
 
+#include <unistd.h>
 #include <iomanip>
 #include <iostream>
 #include <sstream>
--- Makefile.old	2015-02-23 16:33:06.450396216 +0100
+++ Makefile	2015-02-23 16:10:05.721804397 +0100
@@ -264,7 +264,7 @@
 	fi
 
 install: bin/crmc$(SYSTEM)
-	if [[ "$(CRMCROOT)" != "$(PWD)" ]] ; then \
+	if [ "$(CRMCROOT)" != "$(PWD)" ] ; then \
 	 mkdir -p $(CRMCROOT)/bin $(CRMCROOT)/include && \
 	 cp -rf crmc.param tabs lib $(CRMCROOT)/ && \
 	 cp bin/crmc$(SYSTEM) $(CRMCROOT)/bin/crmc && \
