--- src/Makefile.orig	2013-05-28 17:50:13.000000000 +0200
+++ src/Makefile	2013-05-28 17:50:42.000000000 +0200
@@ -53,7 +53,7 @@
 $(LIB_SHARED) : $(objects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) $(FLIBS) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared  $(FLIBS) \
 	$(coreShLibs)
 
 $(LIB_ARCHIVE) : $(objects)
@@ -63,7 +63,7 @@
 $(LIBEXT_SHARED) : $(extobjects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(extobjects) -o $@ -shared -Wl,-soname,$(notdir $@) $(FLIBS) \
+	$(CXX) $(LDFLAGSSHARED) $(extobjects) -o $@ -shared  $(FLIBS) \
 	$(extShLibs)
 
 $(LIBEXT_ARCHIVE) : $(extobjects)
--- src/EvtGenBase/EvtParticle.cpp.orig	2013-11-11 14:27:57.000000000 +0100
+++ src/EvtGenBase/EvtParticle.cpp	2013-11-11 18:09:01.000000000 +0100
@@ -1152,13 +1152,13 @@
     return;
   }
 
-  EvtId idArray[ndaugstore];
-  unsigned int i;
-  for (i = 0; i < ndaugstore; i++) {
-    idArray[i] = idVector[i];
-  }
+  //EvtId idArray(ndaugstore);
+  //unsigned int i;
+  //for (i = 0; i < ndaugstore; i++) {
+  //  idArray[i] = idVector[i];
+  // }
 
-  this->makeDaughters(ndaugstore, idArray);
+  this->makeDaughters(ndaugstore, &idVector[0]);
 
 }
