--- src/Makefile.orig	2013-09-18 13:23:40.000000000 +0200
+++ src/Makefile	2014-01-15 18:10:46.000000000 +0100
@@ -53,7 +53,7 @@
 $(LIB_SHARED) : $(objects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,$(SOFLAGS),$(notdir $@) $(FLIBS) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared  $(FLIBS) \
 	$(coreShLibs)
 
 $(LIB_ARCHIVE) : $(objects)
@@ -63,7 +63,7 @@
 $(LIBEXT_SHARED) : $(extobjects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(extobjects) -o $@ -shared -Wl,$(SOFLAGS),$(notdir $@) $(FLIBS) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared  $(FLIBS) \
 	$(extShLibs)
 
 $(LIBEXT_ARCHIVE) : $(extobjects)
