--- configure.orig	2014-01-18 10:08:50.000000000 +0100
+++ configure	2014-01-18 10:35:11.000000000 +0100
@@ -177,6 +177,7 @@
 CFLAGS="${CFLAGS_OPT}"
 CXXFLAGS="${CXXFLAGS_OPT}"
 FLIBS="-lfrtbegin -lg2c"
+SOFLAGS="-soname"
 
 if [ ${COMPMODE} = OPT ]; then
   FFLAGS="${FFLAGS_OPT}"
@@ -223,6 +224,12 @@
   fi
 fi
 
+# Mac OS platform with gcc4
+if [[ $ARCH == Darwin* ]]; then
+  FLIBS="-lgfortran"
+  SOFLAGS="-install_name"
+fi
+
 #Platform & opt/dbg - independent flags and variables:
 
 echo -n "Creating config.mk ... "
@@ -244,6 +251,7 @@
 CFLAGS = ${CFLAGS}
 CXX = ${CXX}
 CXXFLAGS = ${CXXFLAGS}
+SOFLAGS = ${SOFLAGS}
 
 EVTGENDIR = $(pwd)
 
--- src/Makefile.orig	2014-01-18 10:26:23.000000000 +0100
+++ src/Makefile	2014-01-18 10:29:27.000000000 +0100
@@ -53,7 +53,7 @@
 $(LIB_SHARED) : $(objects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) $(FLIBS) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,$(SOFLAGS),$(notdir $@) $(FLIBS) \
 	$(coreShLibs)
 
 $(LIB_ARCHIVE) : $(objects)
@@ -63,7 +63,7 @@
 $(LIBEXT_SHARED) : $(extobjects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(extobjects) -o $@ -shared -Wl,-soname,$(notdir $@) $(FLIBS) \
+	$(CXX) $(LDFLAGSSHARED) $(extobjects) -o $@ -shared -Wl,$(SOFLAGS),$(notdir $@) $(FLIBS) \
 	$(extShLibs)
 
 $(LIBEXT_ARCHIVE) : $(extobjects)
--- src/EvtGenBase/EvtParticle.cpp.orig	2014-01-23 10:48:33.000000000 +0100
+++ src/EvtGenBase/EvtParticle.cpp	2014-01-23 10:51:58.000000000 +0100
@@ -1154,7 +1154,7 @@
     return;
   }
 
-  EvtId idArray[ndaugstore];
+  EvtId *idArray=new EvtId[ndaugstore];
   unsigned int i;
   for (i = 0; i < ndaugstore; i++) {
     idArray[i] = idVector[i];
@@ -1162,6 +1162,8 @@
 
   this->makeDaughters(ndaugstore, idArray);
 
+  delete[] idArray;
+
 }
 
 void EvtParticle::makeDaughters( unsigned int ndaugstore, EvtId *id){
