--- JHUGenerator/makefile.orig	2013-09-04 18:09:12.000000000 +0200
+++ JHUGenerator/makefile	2013-09-04 18:53:38.000000000 +0200
@@ -1,22 +1,21 @@
-Here = $(PWD)
-ModuleDir = $(Here)/modules
-ObjectDir = $(Here)/objects
-PDFDir = $(Here)/pdfs
-VegasDir = $(Here)/vegas
+ModuleDir = modules
+ObjectDir = objects
+PDFDir = pdfs
+VegasDir = vegas
 
-# compiler options, Comp=ifort/gfort
+# compiler options, Comp=ifort [default] / gfortran
 Opt  = Yes
-Comp = ifort
+Comp?= ifort
 
 
 
-ifeq ($(Comp),gfort)
+ifeq ($(Comp),gfortran)
  ifeq ($(Opt),Yes)
     OptOpts  = -O3
  else
     OptOpts  = -O0
  endif
- fcomp = f95 $(OptOpts) -ffree-line-length-none -I$(VegasDir) -J$(ModuleDir) -Dcompiler=2
+ fcomp = gfortran $(OptOpts) -ffree-line-length-none -I$(VegasDir) -J$(ModuleDir) -Dcompiler=2
 endif
 
 
--- JHUGenME/makefile.orig       2013-09-06 18:41:29.000000000 +0200
+++ JHUGenME/makefile    2013-09-06 18:42:20.000000000 +0200
@@ -1,11 +1,11 @@
-Comp = gfort
+Comp?= gfortran

 ifeq ($(Comp),ifort)
 fcomp = ifort -fpp -O2 -vec-report0 -Dcompiler=1
 # fcomp = ifort -fpp -O0 -vec-report0 -Dcompiler=1
 endif
-ifeq ($(Comp),gfort)
-fcomp = f95 -O3 -ffree-line-length-none -Dcompiler=2
+ifeq ($(Comp),gfortran)
+fcomp = gfortran -O3 -ffree-line-length-none -Dcompiler=2
 endif

 ccomp = gcc
