--- src/Makefile.orig	2013-05-29 20:53:57.000000000 +0200
+++ src/Makefile	2013-05-29 20:54:19.000000000 +0200
@@ -19,7 +19,7 @@
 $(LIB_SHARED):  $(objects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(FC) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@)
+	$(FC) $(LDFLAGSSHARED) $(objects) $(LIBS_HERWIG) -o $@ -shared 
 
 $(LIB_ARCHIVE) : $(objects)
 	mkdir -p $(LIBDIR_ARCHIVE)
