--- bin/lhapdf-getdata.orig	2013-12-16 13:24:03.000000000 +0100
+++ bin/lhapdf-getdata	2013-12-16 13:25:50.000000000 +0100
@@ -34,7 +34,7 @@
         logging.debug(pdflistpage)
         hreq.close()
         import re
-        re_anchor = re.compile(r'^\s*<tr>.*<td><a\s+href="([^"]+)">\1.*$')
+	re_anchor = re.compile(r'^\s*<.*<a\s+href="([^"]+)">\1.*$')
         rtn = []
         for line in pdflistpage.splitlines():
             m = re_anchor.match(line)
@@ -141,6 +141,7 @@
     ## Actually download the sets
     if len(filenames) == 0:
         logging.info("No sets match the arguments given")
+        sys.exit(1)
     else:
         logging.debug("Getting sets " + str(filenames))
         for f in filenames:
@@ -150,4 +151,6 @@
                 logging.info("Forcing overwrite of %s" % setpath)
                 getset = True
             if getset:
-                getPDFSetFile(opts.URL, f, opts.DEST, download=opts.DOWNLOAD)
+                successful = getPDFSetFile(opts.URL, f, opts.DEST, download=opts.DOWNLOAD)
+                if not successful: 
+                    sys.exit(1)
