--- configure.orig	2015-02-17 16:02:27.795756830 +0100
+++ configure	2015-02-17 16:05:28.908237100 +0100
@@ -243,7 +243,7 @@
 configure_package "ROOT" "/afs/cern.ch/sw/lcg/releases/ROOT"\
     "bin" "include" "lib" "rootcint,root-config" "TH1D.h" "Cint" ""
 configure_package "BOOST" "/afs/cern.ch/sw/lcg/releases/Boost"\
-    "bin" "include" "lib" "" "" "boost_iostreams" ""
+    "bin" "include" "lib" "" "" "" ""
 configure_package "GZIP" "" "" "include" "lib" "" "zlib.h" "z" "BOOST"
 
 # Check package dependencies (order does not matter).
