diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Cast/configure CLHEP/Cast/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Cast/configure	2013-03-05 18:04:35.000000000 +0100
+++ CLHEP/Cast/configure	2013-03-05 18:31:21.000000000 +0100
@@ -3112,7 +3112,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Evaluator/configure CLHEP/Evaluator/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Evaluator/configure	2013-03-05 18:04:35.000000000 +0100
+++ CLHEP/Evaluator/configure	2013-03-05 18:31:31.000000000 +0100
@@ -3116,7 +3116,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Exceptions/configure CLHEP/Exceptions/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Exceptions/configure	2013-03-05 18:04:35.000000000 +0100
+++ CLHEP/Exceptions/configure	2013-03-05 18:31:32.000000000 +0100
@@ -3133,7 +3133,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/GenericFunctions/configure CLHEP/GenericFunctions/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/GenericFunctions/configure	2013-03-05 18:04:35.000000000 +0100
+++ CLHEP/GenericFunctions/configure	2013-03-05 18:31:33.000000000 +0100
@@ -3118,7 +3118,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Geometry/configure CLHEP/Geometry/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Geometry/configure	2013-03-05 18:04:36.000000000 +0100
+++ CLHEP/Geometry/configure	2013-03-05 18:31:34.000000000 +0100
@@ -3118,7 +3118,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Matrix/configure CLHEP/Matrix/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Matrix/configure	2013-03-05 18:04:36.000000000 +0100
+++ CLHEP/Matrix/configure	2013-03-05 18:31:35.000000000 +0100
@@ -3118,7 +3118,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Random/configure CLHEP/Random/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Random/configure	2013-03-05 18:04:36.000000000 +0100
+++ CLHEP/Random/configure	2013-03-05 18:31:36.000000000 +0100
@@ -3127,7 +3127,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/RandomObjects/configure CLHEP/RandomObjects/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/RandomObjects/configure	2013-03-05 18:04:36.000000000 +0100
+++ CLHEP/RandomObjects/configure	2013-03-05 18:31:37.000000000 +0100
@@ -3116,7 +3116,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/RefCount/configure CLHEP/RefCount/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/RefCount/configure	2013-03-05 18:04:36.000000000 +0100
+++ CLHEP/RefCount/configure	2013-03-05 18:31:38.000000000 +0100
@@ -3109,7 +3109,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Units/configure CLHEP/Units/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Units/configure	2013-03-05 18:04:36.000000000 +0100
+++ CLHEP/Units/configure	2013-03-05 18:31:39.000000000 +0100
@@ -3115,7 +3115,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Vector/configure CLHEP/Vector/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/Vector/configure	2013-03-05 18:04:36.000000000 +0100
+++ CLHEP/Vector/configure	2013-03-05 18:31:40.000000000 +0100
@@ -3129,7 +3129,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/config/configure CLHEP/config/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/config/configure	2013-03-05 18:04:35.000000000 +0100
+++ CLHEP/config/configure	2013-03-05 18:31:30.000000000 +0100
@@ -3109,7 +3109,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
diff -r -u ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/configure CLHEP/configure
--- ../../../../LocalInstallArea/clhep/1.9.4.7/share/sources/CLHEP/configure	2013-03-05 18:04:35.000000000 +0100
+++ CLHEP/configure	2013-03-05 18:31:28.000000000 +0100
@@ -3095,7 +3095,7 @@
 
 # worry about compiler flags
 case "$CXX" in
-g++)
+*g++*)
    case "$target" in
    *-*-linux*)
       AM_CXXFLAGS="-O -ansi -pedantic -Wall -D_GNU_SOURCE"
--- CLHEP/Random/src/TripleRand.cc.orig	2013-10-28 14:50:27.000000000 +0100
+++ CLHEP/Random/src/TripleRand.cc	2013-10-28 15:00:25.000000000 +0100
@@ -224,13 +224,13 @@
 
 TripleRand::operator float() {
   return (float)
-    ( ( integerCong ^ tausworthe ^ hurd ) * twoToMinus_32 
+    ( ( integerCong ^ tausworthe ^ (unsigned int)hurd ) * twoToMinus_32 
 					+ nearlyTwoToMinus_54 );
 					// make sure non-zero!
 }
 
 TripleRand::operator unsigned int() {
-  return integerCong ^ tausworthe ^ hurd;
+  return integerCong ^ tausworthe ^ (unsigned int)hurd;
 }
 
 Hurd288Engine & TripleRand::Hurd() 	       { return hurd; }
--- CLHEP/Random/src/DRand48Engine.cc.orig 2008-11-13 19:35:53.000000000 +0100
+++ CLHEP/Random/src/DRand48Engine.cc	     2014-03-28 10:39:02.000000000 +0100
@@ -36,6 +36,7 @@
 #include "CLHEP/Random/RandomFunc.h"
 #include "CLHEP/Random/engineIDulong.h"
 #include <string.h>
+#include <cstdlib>
 
 //#define TRACE_IO
 
@@ -62,8 +63,8 @@
    long seed;
    int cycle,curIndex;
 
-   cycle = abs(int(numEngines/maxIndex));
-   curIndex = abs(int(numEngines%maxIndex));
+   cycle = std::abs(int(numEngines/maxIndex));
+   curIndex = std::abs(int(numEngines%maxIndex));
    numEngines += 1;
    long mask = ((cycle & 0x007fffff) << 8);
    HepRandom::getTheTableSeeds( seeds, curIndex );
@@ -77,9 +78,9 @@
    long seed;
    long seeds[2];
 
-   int cycle = abs(int(rowIndex/maxIndex));
-   int row = abs(int(rowIndex%maxIndex));
-   int col = abs(int(colIndex%2));
+   int cycle = std::abs(int(rowIndex/maxIndex));
+   int row = std::abs(int(rowIndex%maxIndex));
+   int col = std::abs(int(colIndex%2));
    long mask = ((cycle & 0x000007ff) << 20);
    HepRandom::getTheTableSeeds( seeds, row );
    seed = (seeds[col])^mask;
