--- src/Makefile.am.orig	2015-02-14 01:40:17.000000001 +0100
+++ src/Makefile.am	2015-02-14 01:44:17.000000001 +0100
@@ -9,7 +9,7 @@
 lib_LTLIBRARIES =
 if !REMOTE_ONLY
 lib_LIBRARIES += libunwind-ptrace.a
-lib_LTLIBRARIES += libunwind.la libunwind-setjmp.la
+lib_LTLIBRARIES += libunwind.la 
 endif
 
 noinst_HEADERS =
@@ -473,6 +473,10 @@
 endif # ARCH_IA64
 endif # ARCH_ARM
 
+if !REMOTE_ONLY
+ lib_LTLIBRARIES += libunwind-setjmp.la
+endif
+
 #
 # Don't link with standard libraries, because those may mention
 # libunwind already.
--- tests/Makefile.am.orig	2011-05-07 05:19:36.000000000 +0200
+++ tests/Makefile.am	2015-02-14 11:09:07.059556494 +0100
@@ -107,7 +107,7 @@
 Gtest_trace_SOURCES = Gtest-trace.c ident.c
 Ltest_trace_SOURCES = Ltest-trace.c ident.c
 
-LIBUNWIND = $(top_builddir)/src/libunwind-$(arch).la
+LIBUNWIND = $(top_builddir)/src/libunwind-$(arch).la $(LIBUNWIND_local)
 LIBUNWIND_ptrace = $(top_builddir)/src/libunwind-ptrace.a
 
 if USE_ELF32

