--- src/gui/kernel/qt_cocoa_helpers_mac_p.h.orig	2014-01-21 15:51:16.000000000 +0100
+++ src/gui/kernel/qt_cocoa_helpers_mac_p.h	2014-01-21 13:58:41.000000000 +0100
@@ -155,8 +155,8 @@
 void qt_dispatchModifiersChanged(void * /*NSEvent * */flagsChangedEvent, QWidget *widgetToGetEvent);
 bool qt_mac_handleTabletEvent(void * /*QCocoaView * */view, void * /*NSEvent * */event);
 inline QApplication *qAppInstance() { return static_cast<QApplication *>(QCoreApplication::instance()); }
-struct ::TabletProximityRec;
-void qt_dispatchTabletProximityEvent(const ::TabletProximityRec &proxRec);
+struct TabletProximityRec;
+void qt_dispatchTabletProximityEvent(const TabletProximityRec &proxRec);
 Qt::KeyboardModifiers qt_cocoaModifiers2QtModifiers(ulong modifierFlags);
 Qt::KeyboardModifiers qt_cocoaDragOperation2QtModifiers(uint dragOperations);
 QPixmap qt_mac_convert_iconref(const IconRef icon, int width, int height);
--- src/declarative/util/qdeclarativefontloader.cpp.orig 2014-03-29 23:40:43.000000000 +0100
+++ src/declarative/util/qdeclarativefontloader.cpp       2014-03-29 23:39:45.000000000 +0100
@@ -64,7 +64,7 @@
 Q_OBJECT
 
 public:
-    QDeclarativeFontObject(int _id);
+    QDeclarativeFontObject(int _id = -1);
 
     void download(const QUrl &url, QNetworkAccessManager *manager);
 
@@ -84,7 +84,7 @@
     Q_DISABLE_COPY(QDeclarativeFontObject)
 };
 
-QDeclarativeFontObject::QDeclarativeFontObject(int _id = -1)
+QDeclarativeFontObject::QDeclarativeFontObject(int _id)
     : QObject(0), id(_id), reply(0), redirectCount(0) {}
 
 
