--- build/macos.gcc.inc.orig	2014-03-29 09:38:17.000000000 +0100
+++ build/macos.gcc.inc		2014-03-29 09:38:33.000000000 +0100
@@ -112,6 +112,6 @@
 # Setting tbbmalloc data.
 #------------------------------------------------------------------------------
 
-M_CPLUS_FLAGS = $(CPLUS_FLAGS) -fno-rtti -fno-exceptions -fno-schedule-insns2
+M_CPLUS_FLAGS = $(CPLUS_FLAGS) -fno-rtti -fno-exceptions 
 
 #------------------------------------------------------------------------------
 # End of setting tbbmalloc data.