--- setupext.py.orig	2013-12-16 15:47:40.000000000 +0100
+++ setupext.py	2013-12-16 15:54:17.000000000 +0100
@@ -58,15 +58,13 @@
     'linux3' : ['/usr/local', '/usr'],
     'linux'  : ['/usr/local', '/usr',],
     'cygwin' : ['/usr/local', '/usr',],
-    '_darwin' : ['/sw/lib/freetype2', '/sw/lib/freetype219', '/usr/local',
+    'darwin' : ['/opt/X11', '/usr/local',
                 '/usr', '/sw'],
     # it appears builds with darwin are broken because of all the
     # different flags the deps can be compile with, so I am pushing
     # people to :
     #   make -f make.osx fetch deps mpl_build mpl_install
 
-    'darwin' : [],
-
     'freebsd4' : ['/usr/local', '/usr'],
     'freebsd5' : ['/usr/local', '/usr'],
     'freebsd6' : ['/usr/local', '/usr'],

--- agg24/include/agg_renderer_outline_aa.h.orig	2014-01-21 09:45:37.000000000 +0100
+++ agg24/include/agg_renderer_outline_aa.h	2014-01-21 09:45:45.000000000 +0100
@@ -1365,7 +1365,7 @@
         //---------------------------------------------------------------------
         void profile(const line_profile_aa& prof) { m_profile = &prof; }
         const line_profile_aa& profile() const { return *m_profile; }
-        line_profile_aa& profile() { return *m_profile; }
+        //line_profile_aa& profile() { return *m_profile; }
 
         //---------------------------------------------------------------------
         int subpixel_width() const { return m_profile->subpixel_width(); }
