--- setup.py
+++ setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python                                                                              
 
 from distutils.core import setup, Extension
 import os, subprocess
@@ -9,9 +9,11 @@
     incdir=incdir.strip('\n')
     libdir=os.popen('root-config --libdir').read()
     libdir=libdir.strip('\n')
+    libs = os.popen('root-config --libdir').read().strip('\n').split()
+    libs.append("Minuit2")
 except RuntimeError:
     print "root-config not found in PATH"
-    
+
 setup(name="pyMinuit2",
       version="0.0.1",
       description="pyMinuit2: Minuit2 interface for minimizing Python functions",
@@ -22,7 +24,6 @@
       ext_modules=[Extension(os.path.join("minuit2"),
                              [os.path.join("minuit2.cpp")],
                              library_dirs=[libdir],
-                             
-libraries=["Core","Cint","RIO","Net","Hist","Graf","Rint","Matrix","MathCore","Minuit2"],
+                             libraries=libs,
                              include_dirs=[incdir]
                              )])
